### ISSUE TITLE
XMPP stream error conditions incorrectly qualified.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2630]
|<img alt="njoyce's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| njoyce reported|
|-|-|
|Trac ID|trac#2630|
|Type|defect|
|Created|2007-05-08 14:23:54Z|

[http://www.xmpp.org/rfcs/rfc3920.html#rfc.section.4.7.2] states that the defined condition of the stream error must be qualified by _urn:ietf:params:xml:ns:xmpp-streams_, but ```t.w.p.j.error.getElement``` is using _urn:ieft:params:xml:ns:xmpp-stanzas_

Additionally, if the ```text``` arg is supplied it too is qualified incorrectly (should be _urn:ietf:params:xml:ns:xmpp-streams_).

<details><summary>Searchable metadata</summary>

```
trac-id__2630 2630
type__defect defect
reporter__njoyce njoyce
priority__highest highest
milestone__Words_0_6 Words-0.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1178634234000000 1178634234000000
changetime__1181289393000000 1181289393000000
version__None None
owner__ 
cc__njoyce cc__ralphm cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm set owner to @ralphm<br>@ralphm set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm removed owner<br>@ralphm set status to `new`|
|-|-|

Fixed in [branches/xmpp-error-2630.](branches/xmpp-error-2630.)

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @ralphm|
|-|-|

The namespace attribute needs some documentation. I'm a bit worried about the break of the Error class (no more namespace), but I guess that's ok. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set owner to @therve|
|-|-|

Added documentation and addressed the other comment by making it explicit that `Error`, now `BaseError` is a base class for `StanzaError` and `StreamError`.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @ralphm|
|-|-|

OK, go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_6"></a>|@ralphm set status to `closed`|
|-|-|

(In [20498]) Use proper namespace for XMPP stream errors.

Author: ralphm
Reviewer: therve
Fixes [#2630](https://github.com/twisted/twisted/issues/2630).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
